packages/lesswrong/server/callbacks/rateLimits.ts

```diff
@@ -78,21 +78,38 @@ async function enforcePostRateLimit (user: DbUser) {
 
 }
 
-async function enforceCommentRateLimit(user: DbUser) {
+
+const userNumberOfCommentsOnOthersPostsInPastTimeframe = async (user: DbUser, hours: number) => {
+  const mNow = moment();
+  const comments = await Comments.find({
+    userId: user._id,
+    createdAt: {
+      $gte: mNow.subtract(hours, 'hours').toDate(),
+    },
+  }).fetch();
+  const postIds = comments.map(comment => comment.postId)
+  const postsNotAuthoredByCommenter = await Posts.find({_id: {$in: postIds}, userId: {$ne:user._id}}).fetch()
```

Comment:
I assume you want to treat post authors the same as post coauthors here? Also since you just need the IDs, I would use a projection.

Reply:
I'm merging this for now, but immediately making a followup PR with these changes – I don't want to go re-test everything again but seems good not to lose track of.